src/App/App.tsx

```diff
@@ -1,11 +1,15 @@
+import RepoPage from "@pages/RepoPage";
 import ReposSearchPage from "@pages/ReposSearchPage";
-import "./App.css";
+import { Routes, Route, Navigate } from "react-router-dom";
 
 function App() {
   return (
-    <div className="container">
-      <ReposSearchPage />
-    </div>
+    <Routes>
+      <Route path="/" element={<Navigate to="/repos" />} />
+      <Route path="/repos" element={<ReposSearchPage />} />
+      <Route path="/repos/:repoId" element={<RepoPage />} />
```

Comment:
здесь возникнет проблема, если придет ссылка /repos/123. Отобразится ReposSearchPage компонент, поскольку роутер будет сравнивать и рендерить по первому вхождению в подстроку, в данном случае, "repos" - подстрока "/repos/:repoId"
чтобы такого не произошло нужно добавить exact, тогда роутер чекнет полное совпадение пути

https://stackoverflow.com/questions/49162311/react-difference-between-route-exact-path-and-route-path

```suggestion
      <Route exact path="/repos" element={<ReposSearchPage />} />
      <Route path="/repos/:repoId" element={<RepoPage />} />
```

Reply:
Я использовал react-router-dom v6, Routes предполагает точное совпадение, exact вроде не нужен, да его и нет вроде, насколько я понял